### PR TITLE
Feature/monster carnival

### PR DESCRIPTION
**Pull request by BenjixD**
## Carnival PQ
- Implemented Monster Carnival 1 & 2
- Added Spieglmann NPC interactions
- Matchmaking allows for 3 minute lobby hold and request to join

### CODE REVIEW

File: scripts/portal/MCrevive6.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+importPackage(Packages.server.partyquest);
+importPackage(Packages.server.partyquest.monstercarnival.components);
+
+function enter(pi) {
```
Comment (nansonzheng) [inline scripts/portal/MCrevive6.js:26]: Why are there 7 nearly identical files about reviving?

File: src/server/life/MobSkill.java [hunk 2]
```diff
@@ -44,7 +44,7 @@
  */
 public class MobSkill {
     private int skillId, skillLevel, mpCon;
-    private List<Integer> toSummon = new ArrayList<Integer>();
+    private List < Integer > toSummon = new ArrayList < Integer > ();
```
Comment (nansonzheng) [inline src/server/life/MobSkill.java:47]: Minor issue: funky spacing with the angle brackets

File: src/server/partyquest/MonsterCarnival.java [hunk 3]
```diff
@@ -1,174 +1,685 @@
...
+            }
+        } else if (team == Team.BLUE) {
+            blueCPLock.readLock().lock();
+            try {
+                return blueTotalCP;
+            } finally {
+                blueCPLock.readLock().unlock();
+            }
+        } else {
+            throw new RuntimeException("Unkown team");
```
Comment (nansonzheng) [inline src/server/partyquest/MonsterCarnival.java]: typo :)

File: src/server/maps/MapleReactorFactory.java [hunk 4]
```diff
@@ -38,6 +38,61 @@
     private static MapleDataProvider data = MapleDataProviderFactory.getDataProvider(new File(System.getProperty("wzpath") + "/Reactor.wz"));
     private static Map<Integer, MapleReactorStats> reactorStats = new HashMap<Integer, MapleReactorStats>();
 
+    
+    public static final MapleReactorStats getReactorS(int rid) {
```
Comment (nansonzheng) [inline src/server/maps/MapleReactorFactory.java:42]: Why are there 2 similarly named functions for getReactor?

File: scripts/portal/MCrevive6.js (hunk 1 above)
Comment (BenjixD) [reply scripts/portal/MCrevive6.js:26]: The WZ files for CPQ1 specify MCRevive 1 - 6. CPQ2 is smarter and only uses MC2Revive.

MCRevive 3 - 4 require you to spawn at specific locations.

**Review by nansonzheng — DISMISSED** [review]

File: src/server/maps/MapleReactorFactory.java (hunk 4 above)
Comment (BenjixD) [reply src/server/maps/MapleReactorFactory.java:42]: Afaik heavenMS uses getRectorS to load in the MCPQ reactors. The old reactor loading function is unchanged.

**Review by nansonzheng — APPROVED** [review]

**Pull request merged**